plugin.py

```diff
@@ -14,3 +19,25 @@ class LspVolarPlugin(NpmClientHandler):
     package_name = __package__
     server_directory = 'server'
     server_binary_path = os.path.join(server_directory, 'node_modules', '@volar', 'server', 'out', 'index.js')
+
+    @classmethod
+    def is_allowed_to_start(
+        cls,
+        window: sublime.Window,
+        initiating_view: Optional[sublime.View] = None,
+        workspace_folders: Optional[List[WorkspaceFolder]] = None,
+        configuration: Optional[ClientConfig] = None
+    ):
+        if not workspace_folders or not configuration:
+            return
+        ts_path = ''
+        resolve_module_script = os.path.join(cls._server_directory_path(), 'resolve_module.js')
```

Comment:
I do not like that I had to use a private method to get the` _server_directory_path`.
Any suggestions?

Reply:
Same for `_node_bin` that I've suggested to use elsewhere.

You could use the public method `cls.get_server` to get the server instead but either way is fine probably.

Reply:
`cls.get_serve` could return `None`. 
I do not want to add an additional if statement. 
I will stick with `_server_directory_path`   :)